pkg/reconciler/instances/istio/action.go

```diff
@@ -201,6 +201,16 @@ func canUninstall(istioVersion actions.IstioVersion) bool {
 	return isInstalled(istioVersion) && istioVersion.ClientVersion != ""
 }
 
+func (h *helperVersion) isDowngradePermitted(second *helperVersion) bool {
```

Comment:
How about introducing a few simple unit tests for this function only? For testing purposes I changed the implementation to the following case:
```go
if second.minor-h.minor == 0 || second.minor-h.minor == 1 || second.minor-h.minor == 2
```
and tests were still green, whereas they should not. Please provide unit tests for this contract.

Reply:
That's right! I will add additional tests for this function.